File: Pinz/Modules/PinzDomain/Sources/Trip.swift
import Foundation
import SwiftUI

public struct Trip: Hashable {
    public var name: String
    public var image: UIImage?
    public var description: String?
    public var pins: [Pin]
    public var season: String?
    public var startDate: String?
    public var endDate: String?
    public var category: String?

    public init(
        name: String,
        image: UIImage? = nil,
        description: String? = nil,
        pins: [Pin],
        season: String? = nil,
        startDate: String? = nil,
        endDate: String? = nil,
        category: String? = nil
    ) {
        self.name = name
        self.image = image
        self.description = description
        self.pins = pins
        self.season = season
        self.startDate = startDate
        self.endDate = endDate
        self.category = category
    }
}

File: Backend/trip-service/internal/models/trip.go
package models

import "time"

// Trip represents a journey (путешествие).
type Trip struct {
	ID            string
	OwnerUserID   string
	Name          string
	Description   string
	Category      string
	Season        string
	Status        string
	PrivacyLevel  string
	StartDate     *time.Time
	EndDate       *time.Time
	LikesCount    int32
	DislikesCount     int32
	MediaCount        int32
	ParticipantsCount int32
	CoverURL          string
	IsPublished   bool
	IsGenerated   bool
	IsSoftDeleted bool
	CreatedAt     time.Time
	UpdatedAt     time.Time
}

// TripParticipant links a user to a trip with optional admin flag.
type TripParticipant struct {
	TripID   string
	UserID   string
	IsAdmin  bool
	JoinedAt time.Time
}

// InvitationLink is a shareable link to join a trip.
type InvitationLink struct {
	ID        string
	TripID    string
	Token     string
	ExpiresAt time.Time
	CreatedAt time.Time
}
